src/Configuration/test/CloudFoundry.Test/CloudfoundryConfigurationProviderTest.cs

```diff
@@ -216,6 +224,96 @@ public void Load_VCAP_APPLICATION_Allows_Reload_Without_Throwing_Exception()
         Assert.Equal("fb8fbcc6-8d58-479e-bcc7-3b4ce5a7f0ca", options.Version);
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task ForwardedHeadersOptions_unrestricted_when_running_on_CloudFoundry(bool isRunningOnCloudFoundry)
+    {
+        using var vcapScope = new EnvironmentVariableScope("VCAP_APPLICATION", isRunningOnCloudFoundry ? "{}" : null);
+        WebApplicationBuilder builder = TestWebApplicationBuilderFactory.CreateDefault();
+        builder.AddCloudFoundryConfiguration();
+        await using WebApplication host = builder.Build();
+
+        ForwardedHeadersOptions options = host.Services.GetRequiredService<IOptions<ForwardedHeadersOptions>>().Value;
+
+        if (isRunningOnCloudFoundry)
+        {
+            options.ForwardedHeaders.Should().HaveFlag(ForwardedHeaders.XForwardedFor);
+            options.ForwardedHeaders.Should().HaveFlag(ForwardedHeaders.XForwardedProto);
+            options.KnownNetworks.Should().BeEmpty();
+            options.KnownProxies.Should().BeEmpty();
+        }
+        else
+        {
+            options.ForwardedHeaders.Should().Be(ForwardedHeaders.None);
```

Comment:
```suggestion
            options.ForwardedHeaders.Should().NotHaveFlag(ForwardedHeaders.XForwardedFor);
            options.ForwardedHeaders.Should().NotHaveFlag(ForwardedHeaders.XForwardedProto);
```
Based on the other conversation, assuming we only care about these two flags.